packages/fxa-auth-server/lib/routes/oauth/index.js

```diff
@@ -298,18 +301,34 @@ module.exports = (log, config, oauthdb, db, mailer, devices) => {
             grantType: request.payload.grant_type,
             uid,
             ecosystemAnonId,
-            clientId: request.payload.client_id,
+            service: request.payload.client_id,
           });
 
           // This is a bit of a hack, but we emit the `account.signed`
           // event to signal to the flow it has been completed (see flowCompleteSignal).
           // Previously, this event would get emitted on the `/certificate/sign`
           // endpoint however, with the move to sync oauth, this does not happen anymore
           // and we need to "fake" it.
-          await request.emitMetricsEvent('account.signed', {
-            uid: uid,
-            device_id: sessionToken.deviceId,
-          });
+          if (scopeSet.contains(OAUTH_SCOPE_OLD_SYNC)) {
+            // The service for this event gets special-cased to 'sync' when the
+            // client is one of the "old" sync clients so that it matches its
+            // pre-oauth `/certificate/sign` event.
+            // ref: https://github.com/mozilla/fxa/pull/6581#issuecomment-702248031
+            const oldSyncClients = ['firefox-desktop', 'firefox-ios'];
+            const service =
+              config.oauth.clientIds &&
+              config.oauth.clientIds[request.payload.client_id] &&
+              oldSyncClients.includes(
+                config.oauth.clientIds[request.payload.client_id]
+              )
+                ? 'sync'
+                : request.payload.client_id;
```

Comment:
We could simplify this by not emitting the event at all if the `service` would not be "sync"; we don't currently have any graphs that use this event for non-sync purposes and we haven't emitted it with any value other than `sync` for a long time. For clarity, I'm imagining something like:

```
if (scopeSet.includes(OLDSYNC) && oldSyncClients.includes(req.payload.client_id)) {
      await request.emitMetricsEvent('account.signed', {
        uid: uid,
        device_id: sessionToken.deviceId,
        service: "sync",
    });
}
```

Again, I don't feel strongly about it, but it might help future folks who find themselves in here if we keep it as minimal as possible.